ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CoreVerifier.kt

```diff
@@ -32,6 +34,24 @@ import org.w3c.dom.Node
 class CoreVerifier(val node: Node) {
     companion object {
         private const val ENCRYPTED_DATA = "EncryptedData"
+
+        /**
+         * Verify SAML extension attributes or elements against the Core Spec document
+         *
+         * 2.4.1.2 Element <SubjectConfirmationData>
+         * 2.7.3.1 Element <Attribute>
+         * 3.2.1 Complex Type RequestAbstractType
+         * 3.2.2 Complex Type StatusResponseType
+         */
+        internal fun verifySamlExtensions(nodes: List<Node>) {
+            nodes.forEach {
+                if (it.namespaceURI == null || it.namespaceURI == TestCommon.SAML_NAMESPACE) {
+                    throw SAMLComplianceException.create(SAMLCore_SamlExtensions,
+                            message = "An invalid SAML extension was found.",
+                            node = it)
+                }
+            }
+        }
```

Comment:
Open to feedback on a better place to put this. Wasn't sure where I should do that.